PokerGame/PokerEngine/Poker.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace PokerEngine
+{
+    public class Poker
+    {
+        public static List<Card> CalculateWinningHand(List<Card> handOne, List<Card> handTwo)
+        {
+            List<Card> winningHand = CaclulateBestPairAndKickerHand(handOne, handTwo);
+            if (winningHand != null)
+            {
+                return winningHand;
+            }
+
+            return CalculateHandWithHighestCardAndKicker(handOne, handTwo);
+        }
+
+        private static List<Card> CaclulateBestPairAndKickerHand(List<Card> handOne, List<Card> handTwo)
+        {
+            if (HasPair(handOne) != Rank.NULL || HasPair(handTwo) != Rank.NULL)
+            {
+                if (HasPair(handOne) > HasPair(handTwo))
+                {
+                    return handOne;
+                }
+                if (HasPair(handOne) < HasPair(handTwo))
+                {
+                    return handTwo;
+                }
+                return handOne;
+            }
+            return null;
+        }
+
+        private static List<Card> CalculateHandWithHighestCardAndKicker(List<Card> handOne, List<Card> handTwo)
+        {
+            List<Card> orderedHandOne = handOne.OrderBy(card => card.rank).ToList();
+            List<Card> orderedHandTwo = handTwo.OrderBy(card => card.rank).ToList();
+
+            for (int i = 0; i < orderedHandOne.Count; i++)
+            {
+                if (orderedHandTwo[i].rank > orderedHandOne[i].rank)
+                {
+                    return handTwo;
+                }
+                if (orderedHandTwo[i].rank < orderedHandOne[i].rank)
+                {
+                    return handOne;
+                }
+            }
+            return null;
+        }
+
+        private static Rank HasPair(List<Card> hand)
```

Comment:
HasPair suggests a bool is returned, but a Rank is returned. This is your NULL problem.  One way to solve this is by using an ```out``` parameter. So ```private static bool HasPair(List<Card> hand, out Card pair)```. Then you can call it with 
``` cs
Card pair;
var hasPair = HasPair(hand, out pair);
if (hasPair)
{
    // do something with pair
}
```

Reply:
ooh okay looks cool. will give this a go :)